content/post/22-03-16-progress-report-31.md

```diff
@@ -0,0 +1,165 @@
+*Contributed by mirrorcult*
+
+Fire extinguishes have been refactored and they can now spray people properly with any reagent, or be refilled from any reagent container.
+
+{{< video-auto "/video/pr_31/firestarter.mp4" >}}
+
+Watch out for homemade flamethrowers!
+
+{{< newsection >}}
+### Misc. Changes
+* Glass tables can now shatter. *Contributed by mirrorcult*
+* Players now move slower. *Contributed by the maintainer team*
+* Disposals pipes can now be constructed. *Contributed by DubiousDoggo*
+* Uplinks can now buy Syndicate Smokes. *Contributed by Rane*
+
+{{< newsection >}}
+## Technical
+
+{{< newsection >}}
+### Multi-grid Stations & Multi-station Maps
+*Contributed by mirrorcult*
+
+Space Station 14 now supports having multiple stations or stations comprised of multiple grids. Gamemodes can also specify what number and combination of maps to load, opening the door for interesting new gamemodes like Station Versus Station.
+
+{{< newsection >}}
+### Server Whitelist
+*Contributed by mirrorcult*
+
+Server hosts can now enable a whitelist using the `whitelist.enabled` cvar. Only players added with the command `whitelistadd <username>` will be able to join, and they can be removed with `whitelistremove <username>`.
+
+If you want to enable it mid-round, you can use the `kicknonwhitelisted` command to give every unwanted player the boot.
+
+{{< newsection >}}
+### Map Renderer
+*Contributed by DrSmugleaf, moony, and mirrorcult*
+
+`Content.MapRenderer` has been added as a new project that can take a map file and render it as an image. It's extremely useful for development, updating the wiki, and more.
+
+{{< imgw "images/post/pr_31/bagelrender" >}}
+
+{{< newsection >}}
+## Credits
+### Contributors Since Last Progress Report
+*354 commits, authored by: metalgearsloth, Peptide90, Rane, Pancake, Kara D, ShadowCommander, Moony, Lamrr, mirrorcult, wrexbe, Pieter-Jan Briers, Vera Aguilera Puerto, LittleBuilderJane, Alex Evgrashin, Michael Phillips, Morbo, Sam Weaver, Daemon, Delete69, ElectroJr, DubiousDoggo, Júlio César Ueti, Paul, ScalyChimp, BubblegumBlue, Charlese2, DrSmugleaf, E F R, Fishfish458, T-Stalker, Zymem, Aaron Mell, Acruid, Clyybber, Demetre Beroshvili, Flipp Syder, Julian Giebel, JustinTime, Matz05, Neith, Plasmaguy, TaralGit, Willhelm53, WlarusFromDaSpace, Ygg01, bloodrizer, keronshb, pointer-to-null, router, sewerpig*
+
+### Patrons
+*Aaron Mell, Acruid, Alex Evgrashin, BubblegumBlue, Charlese2, Clyybber, Daemon, Delete69, Demetre Beroshvili, DrSmugleaf, E F R, ElectroJr, Fishfish458, Flipp Syder, DubiousDoggo, Julian Giebel, JustinTime, Júlio César Ueti, Kara D, Lamrr, LittleBuilderJane, Matz05, Michael Phillips, Moony, Morbo, Neith, Pancake, Paul, Peptide90, Pieter-Jan Briers, Plasmaguy, Rane, Sam Weaver, ScalyChimp, ShadowCommander, T-Stalker, TaralGit, Vera Aguilera Puerto, Willhelm53, WlarusFromDaSpace, Ygg01, Zymem, bloodrizer, keronshb, metalgearsloth, mirrorcult, pointer-to-null, router, sewerpig, wrexbe*
```

Comment:
List of Patrons is wrong, I'm not one of them and Tomeno is missing for example.

Reply:
Done in [`d33a7e6` (#27)](https://github.com/space-wizards/website-content/pull/27/commits/d33a7e6311c71bc1065c548a98b3b044ea77e021)